packages/block-library/src/preformatted/index.js

```diff
@@ -68,7 +68,7 @@ export const settings = {
 		return (
 			<RichText
 				tagName="pre"
-				value={ content }
+				value={ content.replace( /\n/g, '<br>' ) }
```

Comment:
Does this mean we should do the opposite once `onChange` is called?
Are there other tags that needs to behave this way? Should this be a config of `RichText`?

Reply:
No, we don't need modify `onChange`. There are two possibilities: `<br>` and `\n`. In `RichText` `\n` is collapsed as it is normally in HTML, so it cannot handle it. I went for using `<br>` by default, but we could go for `\n` too, in which case we would need to `onChange`. RichText always outputs `<br>` for line breaks.

Yes, it could also be part of the `RichText` config. Something like `collapseLineBreaks: false`. Not sure if that's needed as it will be more of an exception and it can be resolved outside of `RichText`, keeping `RichText` free of more parameters.

Reply:
> Yes, it could also be part of the `RichText` config. Something like `collapseLineBreaks: false`.

Does it need to be configured? Or is it enough to infer by `tagName === 'pre'` ?

Reply:
Why does RichText collapse HTML anyways? From the comment in `filterString`, it reads as though it's not strictly necessary, but more to the point of normalizing on an assumption it'd be treated as equivalent by the browser anyways. If so, wouldn't this pull request and corresponding issue contradict that assumption?

https://github.com/WordPress/gutenberg/blob/910b655018037167c2dcee38095905003f632b51/packages/rich-text/src/create.js#L235-L239

Reply:
* Without removing line breaks, the line breaks will be visible in `RichText`, just like `<br>`. So if you add a line break in the HTML editor, then switch back, the line break will remain visible. 
* If we were to keep textual line breaks, they'd interfere with the rich text value line breaks:
  * In HTML, `\n` means nothing, it's just for the formatting of HTML, `<br>` means line break.
  * In a rich text value, `\n` means line break.

So if we just leave `\n`, they'll all be serialised as line breaks.